File: Forms/About.vb
' Utility to automatically download radio programmes, using a plugin framework for provider specific implementation.
' Copyright © 2007-2010 Matt Robinson
'
' This program is free software; you can redistribute it and/or modify it under the terms of the GNU General
' Public License as published by the Free Software Foundation; either version 2 of the License, or (at your
' option) any later version.
'
' This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the
' implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
' License for more details.
'
' You should have received a copy of the GNU General Public License along with this program; if not, write
' to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

Option Strict On
Option Explicit On

Imports System.Diagnostics.Process

Friend NotInheritable Class About
    Private Sub About_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        ' Set the title of the form.
        Dim ApplicationTitle As String
        If My.Application.Info.Title <> "" Then
            ApplicationTitle = My.Application.Info.Title
        Else
            ApplicationTitle = System.IO.Path.GetFileNameWithoutExtension(My.Application.Info.AssemblyName)
        End If

        Me.Font = SystemFonts.MessageBoxFont

        Me.Text = "About " + ApplicationTitle
        Me.LabelNameAndVer.Text = My.Application.Info.ProductName + " " + My.Application.Info.Version.ToString
        Me.LabelCopyright.Text = My.Application.Info.Copyright
    End Sub

    Private Sub OKButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ButtonOK.Click
        Me.Close()
    End Sub

    Private Sub HomepageLink_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles HomepageLink.Click
        Start("http://www.nerdoftheherd.com/tools/radiodld/")
    End Sub
End Class

File: Forms/About.cs
/*
 * Copyright © 2007-2015 Matt Robinson
 *
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

namespace RadioDld
{
    using System;
    using System.Drawing;
    using System.Windows.Forms;

    internal sealed partial class About : Form
    {
        public About()
        {
            this.InitializeComponent();
        }

        private void About_Load(object sender, EventArgs e)
        {
            this.Font = SystemFonts.MessageBoxFont;

            this.Text = "About " + Application.ProductName;
            this.LabelNameAndVer.Text = Application.ProductName + " " + Application.ProductVersion;
            this.LabelCopyright.Text = new Microsoft.VisualBasic.ApplicationServices.ApplicationBase().Info.Copyright;

            UpdateCheck.CheckAvailable(this.UpdateAvailable);
        }

        private void UpdateAvailable()
        {
            if (this.InvokeRequired)
            {
                this.Invoke((MethodInvoker)(() => { this.UpdateAvailable(); }));
                return;
            }

            this.LinkUpdate.Visible = true;
        }

        private void ButtonOk_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void LinkHomepage_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            OsUtils.LaunchUrl(new Uri(this.LinkHomepage.Text), "About Dialog Link");
        }

        private void LinkUpdate_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            using (UpdateNotify showUpdate = new UpdateNotify())
            {
                if (showUpdate.ShowDialog(this) == DialogResult.Yes)
                {
                    OsUtils.LaunchUrl(new Uri("https://nerdoftheherd.com/tools/radiodld/"), "Download Update (About)");
                    this.Close();
                }
            }
        }
    }
}
